api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -0,0 +1,116 @@
+package org.pmiops.workbench.api;
+
+import org.pmiops.workbench.db.dao.UserRecentResourceService;
+import org.pmiops.workbench.db.dao.WorkspaceService;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.UserRecentResource;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.model.Cohort;
+import org.pmiops.workbench.model.FileDetail;
+import org.pmiops.workbench.model.RecentResource;
+import org.pmiops.workbench.model.RecentResourceResponse;
+import org.pmiops.workbench.firecloud.model.WorkspaceResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.inject.Provider;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@RestController
+public class UserMetricsController implements UserMetricsApiDelegate {
+   Provider<User> userProvider;
+  UserRecentResourceService userRecentResourceService;
+  WorkspaceService workspaceService;
+  FireCloudService fireCloudService;
+  Map<Long, String> workspaceAccessMap = new HashMap<Long, String>();
```

Comment:
A controller instance is shared across requests, there should never be any request-scoped variables stored at this level, i.e. it may contain data for a completely different user or be concurrently modified on access.